### ISSUE TITLE
Dependency issue starting with new rails project

### ISSUE THREAD

**Issue by ghost** [relevance/tarantula#1]
After following the instructions on the the github page I could not get it to run on a new rails app.

here is what I tried

$ rails spider-app (i am using 2.3.2 for this) 
$ gem install relevance-tarantula—source http://gems.github.com

added this to my test.rb at the bottom config.gem ‘relevance-tarantula’, :source => “http://gems.github.com”, :lib => ‘relevance/tarantula’

$ mkdir -p vendor/gems 
$ cd vendor/gems 
$ sudo gem unpack relevance-tarantula

added this to the bottom of the rake file load File.join(RAILS_ROOT, Dir[“vendor/gems/relevance-tarantula-/tasks/.rake”])

$ rake tarantula:setup 
$ rake tarantula:test

then I get this error: ... /Users/Tony/.gem/ruby/1.8/gems/rake-0.8.3/lib/rake/rake_test_loader.rb:5:in `each’ /Users/Tony/.gem/ruby/1.8/gems/rake-0.8.3/lib/rake/rake_test_loader.rb:5 Missing these required gems: relevance-tarantula

You’re running: ruby 1.8.6.114 at /System/Library/Frameworks/Ruby.framework/Versions/1.8/usr/bin/ruby rubygems 1.3.1 at /Users/Tony/.gem/ruby/1.8, /Library/Ruby/Gems/1.8, /System/Library/Frameworks/Ruby.framework/Versions/1.8/usr/lib/ruby/gems/1.8

Run `rake gems:install` to install the missing gems. rake aborted! Command failed with status (1): [/System/Library/Frameworks/Ruby.framework/...]

I can see the gem installed both in vendor and locally when I do a gem list -l tarantula.

[original LH ticket](http://relevance.lighthouseapp.com/projects/17868/tickets/17)

 This ticket has 0 attachment(s).

**Comment by ghost** [comment]
**Dependency issue starting with new rails project**

I dug a little deeper and found out that the missing required gem was htmlentities.

I just needed to look at the top of the stack instead of the one at the bottom.

no such file to load -- htmlentities
/Library/Ruby/Site/1.8/rubygems/custom_require.rb:31:in `gem_original_require&rsquo;
/Library/Ruby/Site/1.8/rubygems/custom_require.rb:31:in`require&rsquo;
/Library/Ruby/Gems/1.8/gems/activesupport-2.3.2/lib/active_support/dependencies.rb:158:in `require&rsquo;
/Users/Tony/dev/rails/spider/vendor/gems/relevance-tarantula-0.1.7/lib/relevance/tarantula.rb:14
/Library/Ruby/Site/1.8/rubygems/custom_require.rb:31:in`gem_original_require&rsquo;
/Library/Ruby/Site/1.8/rubygems/custom_require.rb:31:in `require&rsquo;
/Library/Ruby/Gems/1.8/gems/activesupport-2.3.2/lib/active_support/dependencies.rb:158:in`require&rsquo;
/Library/Ruby/Gems/1.8/gems/rails-2.3.2/lib/rails/gem_dependency.rb:179:in `load&rsquo;
/Library/Ruby/Gems/1.8/gems/rails-2.3.2/lib/initializer.rb:305:in`load_gems&rsquo;
/Library/Ruby/Gems/1.8/gems/rails-2.3.2/lib/initializer.rb:305:in `each&rsquo;
/Library/Ruby/Gems/1.8/gems/rails-2.3.2/lib/initializer.rb:305:in`load_gems&rsquo;
/Library/Ruby/Gems/1.8/gems/rails-2.3.2/lib/initializer.rb:167:in `process&rsquo;
/Library/Ruby/Gems/1.8/gems/rails-2.3.2/lib/initializer.rb:113:in`send&rsquo;
/Library/Ruby/Gems/1.8/gems/rails-2.3.2/lib/initializer.rb:113:in `run&rsquo;
/Users/Tony/dev/rails/spider/config/environment.rb:9
./test/tarantula/../test_helper.rb:2:in`require&rsquo;
./test/tarantula/../test_helper.rb:2
./test/tarantula/tarantula_test.rb:1:in `require&rsquo;
./test/tarantula/tarantula_test.rb:1
/Library/Ruby/Gems/1.8/gems/rake-0.8.4/lib/rake/rake_test_loader.rb:5:in`load&rsquo;
/Library/Ruby/Gems/1.8/gems/rake-0.8.4/lib/rake/rake_test_loader.rb:5
/Library/Ruby/Gems/1.8/gems/rake-0.8.4/lib/rake/rake_test_loader.rb:5:in `each&rsquo;
/Library/Ruby/Gems/1.8/gems/rake-0.8.4/lib/rake/rake_test_loader.rb:5
Missing these required gems:
  relevance-tarantula

 by Tony Eichelberger

**Comment by ghost** [comment]
**Dependency issue starting with new rails project**

Thanks for reporting this issue.  We&rsquo;ll find a clean Linux installation (i.e., with no gems pre-installed) and see if we can recreate this issue.  If we&rsquo;re able to reproduce it, we&rsquo;ll look into updating Tarantula to properly declare its dependencies.

 by Jason Rudolph

**Comment by jasonrudolph** [comment]
This issue is caused by the inability for RubyGems to resolve dependencies from Rubyforge when you specifically list GitHub as the source for the installation.

```
$ gem install relevance-tarantula --source http://gems.github.com
WARNING:  Installing to ~/.gem since /usr/lib/ruby/gems/1.8 and
      /usr/bin aren't both writable.
ERROR:  Error installing relevance-tarantula:
    relevance-tarantula requires htmlentities (>= 0, runtime)
```

## SOLUTION 1

Install from Rubyforge

```
$ gem install tarantula
WARNING:  Installing to ~/.gem since /usr/lib/ruby/gems/1.8 and
      /usr/bin aren't both writable.
Building native extensions.  This could take a while...
Successfully installed htmlentities-4.0.0
Successfully installed hpricot-0.8.1
Successfully installed tarantula-0.1.8
3 gems installed
Installing ri documentation for htmlentities-4.0.0...
Installing ri documentation for hpricot-0.8.1...
Installing ri documentation for tarantula-0.1.8...
Installing RDoc documentation for htmlentities-4.0.0...
Installing RDoc documentation for hpricot-0.8.1...
Installing RDoc documentation for tarantula-0.1.8...
```

## SOLUTION 2

Add GitHub as a source for your RubyGems installation

```
$ gem sources -a http://gems.github.com
http://gems.github.com added to sources
$  gem install relevance-tarantula
WARNING:  Installing to ~/.gem since /usr/lib/ruby/gems/1.8 and
      /usr/bin aren't both writable.
Building native extensions.  This could take a while...
Successfully installed htmlentities-4.0.0
Successfully installed hpricot-0.8.1
Successfully installed relevance-tarantula-0.1.8
3 gems installed
Installing ri documentation for htmlentities-4.0.0...
Installing ri documentation for hpricot-0.8.1...
Installing ri documentation for relevance-tarantula-0.1.8...
Installing RDoc documentation for htmlentities-4.0.0...
Installing RDoc documentation for hpricot-0.8.1...
Installing RDoc documentation for relevance-tarantula-0.1.8...
```

**Comment by jasonrudolph** [comment]
Update README with better instructions for installing from GitHub. Closed by d492625751989e06f11701acc3045276e0a5f281.
